src/simsopt/util/dipole_array_helper_functions.py

```diff
@@ -0,0 +1,754 @@
+"""
+This module contains the a number of useful functions for  
+optimizing dipole arrays in the SIMSOPT code.
+"""
+__all__ = ['remove_inboard_dipoles',
+           'remove_interlinking_dipoles_and_TFs',
+           'align_dipoles_with_plasma',
+           'initialize_coils',
+           'dipole_array_optimization_function',
+           'save_coil_sets',
+           'quaternion_from_axis_angle',
+           'quaternion_multiply',
+           'rotate_vector',
+           'compute_quaternion',
+           'compute_fourier_coeffs',
+           'rho_fourier',
+           'generate_even_arc_angles',
+           'generate_windowpane_array',
+           'generate_tf_array',
+           'generate_curves',
+           'rho',
+           'a_m',
+           'b_m'
+           ]
+
+import numpy as np
+
+
+def remove_inboard_dipoles(plasma_surf, base_curves, eps=-0.4):
+    """
+    Remove all the dipole coils on the inboard side. Useful if the desired plasma
+    configuration is fairly compact, so no room for dipoles on inboard side.
+
+    Args:
+        plasma_surf: Surface object
+            The plasma boundary surface.
+        base_curves:
+            The curve objects for the dipoles in the array.
+        eps: float
+            Controls how inboard a dipole can be without being removed.
+
+    Returns:
+        base_curves:
+            The same objects, minus any dipoles on the inboard side. 
+    """
+    import warnings
+    keep_inds = []
+    for ii in range(len(base_curves)):
+        counter = 0
+        for i in range(base_curves[0].gamma().shape[0]):
+            dij = np.sqrt(np.sum((base_curves[ii].gamma()[i, :]) ** 2))
+            conflict_bool = (dij < (1.0 + eps) * plasma_surf.get_rc(0, 0))
+            if conflict_bool:
+                print('bad index = ', i, dij, plasma_surf.get_rc(0, 0))
+                warnings.warn(
+                    'There is a PSC coil initialized such that it is within a radius'
+                    'of a TF coil. Deleting these PSCs now.')
+                counter += 1
+                break
+        if counter == 0:
+            keep_inds.append(ii)
+    return np.array(base_curves)[keep_inds]
+
+
+def remove_interlinking_dipoles_and_TFs(base_curves, base_curves_TF, eps=0.05):
+    """
+    Similar process to remove any dipole coils that are initialized
+    intersecting with the TF coils, to avoid interlinking at the very beginning of optimization.
+
+    Args:
+        base_curves:
+            The curve objects for the dipoles in the array.
+        base_curves_TF:
+            The curve objects for the TF (modular) coils in the array.
+        eps: float
+            controls how close a TF-dipole coil distance can be before being removed.
```

Comment:
same comment as above, can the docstring reflect how interlinking coils are detected + what is eps?